src/main/scala/wandou/avpath/Evaluator.scala

```diff
@@ -376,26 +403,27 @@ object Evaluator {
       case null =>
         // select record itself
         ctxs foreach {
-          case Ctx(rec: IndexedRecord, schema, topLevelField, _) =>
-            res ::= Ctx(rec, schema, null, Some(TargetRecord(rec, null)))
+          case Ctx(rec: IndexedRecord, name, schema, topLevelField, path, _) =>
+            res ::= Ctx(rec, name, schema, null, path, Some(TargetRecord(rec, null)))
           case _ => // should be rec
         }
       case "*" =>
         ctxs foreach {
-          case Ctx(rec: IndexedRecord, schema, topLevelField, _) =>
+          case Ctx(rec: IndexedRecord, name, schema, topLevelField, path, _) =>
             val fields = rec.getSchema.getFields.iterator
             while (fields.hasNext) {
               val field = fields.next
               val value = rec.get(field.pos)
-              res ::= Ctx(value, field.schema, if (topLevelField == null) field else topLevelField, Some(TargetRecord(rec, field)))
+              res ::= Ctx(value, field.name, field.schema, if (topLevelField == null) field else topLevelField, path, Some(TargetRecord(rec, field)))
             }
-          case Ctx(arr: java.util.Collection[_], schema, topLevelField, _) =>
+          case Ctx(arr: java.util.Collection[_], name, schema, topLevelField, path, _) =>
             getElementType(schema) foreach { elemType =>
               val values = arr.iterator
               var j = 0
               while (values.hasNext) {
                 val value = values.next
-                res ::= Ctx(value, elemType, topLevelField, Some(TargetArray(arr, j, schema)))
+                // not sure
```

Comment:
We are on a list, we name each sub element like the parent.
It should not be troublesome, but it's not right.